### PR TITLE
v6 — fail safe

### LINKED ISSUE #26: Deprecate `isSwitchedOff`.
I would like to deprecate `isSwitchedOff`.

I don't _think_ it's possible in JavaScript yet (coming soon in ES6 with [Proxies](http://stackoverflow.com/questions/7891937/is-it-possible-to-implement-dynamic-getters-setters-in-javascript)) to do:-

```js
flags.undefinedFlag.isSwitchedOff === true
```

And I quite like being able to make template-level decisions on feature flags without having to use custom helpers — and negative if is really easy…

```ms
{{^flags.undefinedFlag.isSwitchedOn}}
{{/flags.undefinedFlag.isSwitchedOn}}
```

// @commuterjoy @wheresrhys

### PR DESCRIPTION
**Pull request by matthew-andrews**
Closes #26

### CODE REVIEW

File: lib/flags.js [hunk 1]
```diff
@@ -55,44 +51,43 @@ Flags.prototype.setUrl = function (url) {
...
 	if (flagname) {
-		return (this.flags[flagname] || new Flag({
-			name: flagname + ' (unknown)',
-			description: 'unknown',
-			safeState: false,
-			expiry: 0
-		}, {
-			techDebtAlerter: this.techDebtAlerter
-		}));
+		flag = this.flags[flagname];
```
Comment (wheresrhys) [inline lib/flags.js]: don't need this line?

Comment (matthew-andrews) [inline lib/flags.js]: oh yeah

**Comment by wheresrhys** [comment]
:+1: For the substance of the change and the implementation. My only reservation is roll-out - updating flags across all our components and apps is a mammoth task. Maybe best left till after the alpha deadline when we have a bit of breathing space to deal with tech debt? I've got no strong opinion either way, just a  little nervous

**Comment by matthew-andrews** [comment]
yah you're right. :(

**Pull request merged**